### PR TITLE
Try running tests on multiple platforms

### PR DESCRIPTION
**Pull request by oli-obk**
work towards https://github.com/rust-lang/miri/issues/2724

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -92,6 +92,25 @@ impl Default for Config {
...
+            .canonicalize()
+            .unwrap()
+            .parent()
+            .unwrap()
+            .display()
+            .to_string()
+            // Escape windows path backslashes so that regexes see the plain backslashes instead
+            // of escaping the follow-up symbol.
+            .replace('\\', r#"\\"#);
+        self.stderr_filter(pattern, replacement);
```
[the remark this answers is not in the document]
Comment (RalfJung) [reply src/lib.rs:111]: Oh I see, this is an even lower-level issue -- we are using regex normalization for the `$DIR` thing and that was going wrong.

Is there no function to turn a string into a regex that matches exactly that string, applying suitable escaping?

[the remark this answers is not in the document]
Comment (oli-obk) [reply src/lib.rs:111]: There is not, but I switched to just matching the vec exactly

File: src/lib.rs [hunk 2]
```diff
@@ -169,7 +181,36 @@ pub enum OutputConflictHandling {
     Bless,
 }
 
-pub type Filter = Vec<(Regex, &'static [u8])>;
+/// A filter's match rule.
+#[derive(Debug)]
+pub enum Match {
+    /// If the regex matches, the filter applies
+    Regex(Regex),
+    /// If the path is found, the filter applies
```
Comment (RalfJung) [inline src/lib.rs:189]: "path"? There is no path here. Should probably be "if that exact sequence of bytes is found"?

File: src/lib.rs [hunk 3]
```diff
@@ -74,7 +75,7 @@ impl Default for Config {
             trailing_args: vec![],
             host: None,
             target: None,
-            stderr_filters: vec![],
+            stderr_filters: vec![(Match::Exact(vec![b'\\']), b"/")],
```
Comment (RalfJung) [inline src/lib.rs:78]: Hard-coding a default filter is kind of suspicious. Why is this needed?

File: src/lib.rs [hunk 4]
```diff
@@ -92,13 +93,24 @@ impl Default for Config {
 }
 
 impl Config {
+    /// Replace a path with a pattern if it appears in stderr.
```
Comment (RalfJung) [inline src/lib.rs:96]: It's replaced with `replacement`, not with a pattern, right?

```suggestion
    /// Add filter to replace given path if it appears in stderr.
```

File: src/lib.rs (hunk 2 above)
Comment (oli-obk) [reply src/lib.rs:189]: ah yea, I had a `PathBuf` in there earlier

File: src/lib.rs (hunk 3 above)
Comment (oli-obk) [reply src/lib.rs:78]: it's necessary, but not sufficient. This just gets rid of all the backslashes in stderr files without having to repeat that filter everywhere

Comment (RalfJung) [reply src/lib.rs:78]: I'm a bit worried about having hard-coded default filters without an opt-out.

Comment (oli-obk) [reply src/lib.rs:78]: we have an opt out, you can just overwrite the `stderr_filters` field, it's public.

Comment (RalfJung) [reply src/lib.rs:78]: Oh I see. Sounds good then.

Comment (RalfJung) [reply src/lib.rs:78]: But this should be documented somewhere.

**Comment by oli-obk** [comment]
It works @RalfJung
